third_party/rust/chromium_crates_io/vendor/orchard-0.7.1/src/zip32.rs

```diff
@@ -122,7 +122,7 @@ impl KeyIndex {
 ///
 /// [orchardextendedkeys]: https://zips.z.cash/zip-0032#orchard-extended-keys
 #[derive(Debug, Clone)]
-pub(crate) struct ExtendedSpendingKey {
+pub struct ExtendedSpendingKey {
```

Comment:
Any idea why this isn't  part of the public API? Has upstream indicated whether they'd take the patch?

Reply:
They have [SpendingKey](https://github.com/zcash/orchard/blob/2b9c9a1deb66f8b20cd5c07fdd0cec87895f5c16/src/keys.rs#L93) which matches exact account.
We can reuse it but we would need to keep seed in the ZCashKeyring.
I am not sure this is a good idea from security view. Haven't seen such usages before.
So to match common behavior where we keep some master key in the keyring we need to make this thing public.

Reply:
We can't store seed in memory, it is encrypted by user password and stored in prefs. When we need to use it, user password is required to decrypt it, ex revealing seed phrase.

Reply:
Copying a Slack thread here:
> Oleg: What is the reason of why [ESK](https://github.com/zcash/orchard/blob/2b9c9a1deb66f8b20cd5c07fdd0cec87895f5c16/src/zip32.rs#L125) is not a part of public api?
> @str4d: Because there is basically no reason for it to be used directly. Instead use `SpendingKey::from_zip32_seed`.
> Oleg: I am asking because in our case we don't keep seed in the memory instead we keep master key to generate derived accounts. So we had to patch `ExtendedSpendingKey`.
> @str4d: Keeping the master key around in memory is just as risky as keeping the seed in memory, so I don't see why you wouldn't just load it when needed to generate an account or to spend, and otherwise only keep the account public keys in memory.

Reply:
Just pasting the clarification from @darkdh here:
@darkdh: To clarify, we don’t keep master key(derived from seed directly) nor seed in memory. What we keep is each private key derived from master key according to each path per coin.